src/UI/App.jsx

```diff
@@ -39,28 +74,237 @@ export default class App extends React.Component {
 			undefined,
 			undefined,
 			DeviceTypes.Web,
-			version,
+			this.state.deviceVersion,
 		);
 		this.messageIDs = new Set();
+		this.visibilityChange = "visibilitychange";
+
+		// Make sure layers to proxy exist
+		window.parleySettings
+			= window.parleySettings ? window.parleySettings : {};
+		window.parleySettings.runOptions
+			= window.parleySettings.runOptions ? window.parleySettings.runOptions : {};
+		window.parleySettings.runOptions.interfaceTexts
+			= window.parleySettings.runOptions.interfaceTexts ? window.parleySettings.runOptions.interfaceTexts : {};
+	}
+
+	// eslint-disable-next-line no-unused-vars
+	shouldComponentUpdate(nextProps, nextState, nextContext) {
```

Comment:
Is het niet logischer om dit in `getDerivedStateFromProps` te doen dan in `shouldComponentUpdate`?

Reply:
Hmm dit is een lastige..

`getDerivedStateFromProps` zegt
> getDerivedStateFromProps exists for only one purpose. It enables a component to update its internal state as the result of changes in props. ... as a general rule, derived state should be used sparingly

https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

`shouldComponentUpdate` zegt
> Use shouldComponentUpdate() to let React know if a component’s output is not affected by the current change in state or props. The default behavior is to re-render on every state change, and in the vast majority of cases you should rely on the default behavior.

En beiden zijn meer gericht op props en een conditionele return waarde.
Maar wij zijn meer opzoek naar een manier om te luisteren naar `state` changes, dus misschien is `componentDidUpdate` beter..

`componentDidUpdate` zegt
> Use this as an opportunity to operate on the DOM when the component has been updated. This is also a good place to do network requests as long as you compare the current props to previous props (e.g. a network request may not be necessary if the props have not changed).

Alleen probleem met deze is weer dat ie ná de render gedaan wordt. 
![image](https://user-images.githubusercontent.com/26112463/163138888-238d6065-d001-474a-bf28-d01a5935e86f.png)

Dus als we bv deze uitvoeren http://github.com/parley-messaging/web-library/blob/15e18dd4682cf073ebcbd3bf402caf2684be48cf/src/UI/App.jsx#L94-L95 dan krijgt ie weer een nieuwe render omdat die weer de state gaat updaten http://github.com/parley-messaging/web-library/blob/15e18dd4682cf073ebcbd3bf402caf2684be48cf/src/UI/App.jsx#L226-L246

Dus eindigen we met 1 extra render.. Dan lijkt `shouldComponentUpdate` toch beter..

Reply:
Hmm in beide gevallen doet ie even veel renders..
Dus dan is `componentDidUpdate()` prima denk ik

![image](https://user-images.githubusercontent.com/26112463/163576486-19719b4a-f4b7-4b21-83f2-a4c4910c16d8.png)

Reply:
Done